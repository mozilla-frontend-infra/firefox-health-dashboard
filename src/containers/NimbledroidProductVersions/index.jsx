import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import fetchNimbledroidData from '../../utils/nimbledroid/fetchNimbledroidData';
import CONFIG from '../../utils/nimbledroid/config';
import { withErrorBoundary } from '../../vendor/errors';

const styles = {
  root: {},
  spacing: {
    paddingRight: '0.5rem',
  },
};

class NimbledroidProductVersions extends Component {
  state = {
    meta: null,
  };

  async componentDidMount() {
    const { nimbledroidData, products } = this.props;
    const { meta } = nimbledroidData || (await fetchNimbledroidData(products));

    this.setState({ meta });
  }

  render() {
    const { classes } = this.props;
    const { meta } = this.state;

    return !meta ? null : (
      <div className={classes.root}>
        {Object.keys(meta).map(packageId => (
          <span key={packageId} className={classes.spacing}>
            {`${CONFIG.packageIdLabels[packageId]} (${
              meta[packageId].latestVersion
            })`}
          </span>
        ))}
      </div>
    );
  }
}

NimbledroidProductVersions.propTypes = {
  classes: PropTypes.shape({}),
  nimbledroidData: PropTypes.shape({}),
  products: PropTypes.arrayOf(PropTypes.string),
};

export default withErrorBoundary(
  withStyles(styles)(NimbledroidProductVersions)
);
