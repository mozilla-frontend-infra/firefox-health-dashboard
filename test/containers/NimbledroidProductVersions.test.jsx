import React from 'react';
import renderer from 'react-test-renderer';
import NimbledroidProductVersions from '../../src/containers/NimbledroidProductVersions';

const nimbledroidData = require('../mocks/nimbledroidData.json');

it('renders correctly', () => {
  const tree = renderer
    .create(<NimbledroidProductVersions nimbledroidData={nimbledroidData} />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
