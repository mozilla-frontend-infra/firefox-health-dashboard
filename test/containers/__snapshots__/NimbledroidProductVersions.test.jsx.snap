// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="NimbledroidProductVersions-root-1"
>
  <span
    className="NimbledroidProductVersions-spacing-2"
  >
    GeckoView (8.0)
  </span>
  <span
    className="NimbledroidProductVersions-spacing-2"
  >
    WebView (8.0)
  </span>
  <span
    className="NimbledroidProductVersions-spacing-2"
  >
    ChromeBeta (70.0.3538.17)
  </span>
</div>
`;
